honeycomb-core/src/twomap.rs

```diff
@@ -922,45 +366,174 @@ impl<T: CoordsFloat> CMap2<T> {
     /// - `betas: [DartIdentifier; 3]` -- Value of the images as
     ///   *[β<sub>0</sub>(dart), β<sub>1</sub>(dart), β<sub>2</sub>(dart)]*
     ///
-    /// # Example
-    ///
-    /// See [CMap2] example.
-    ///
     pub fn set_betas(&mut self, dart_id: DartIdentifier, betas: [DartIdentifier; CMAP2_BETA]) {
         self.betas[dart_id as usize] = betas;
     }
+}
 
-    /// Set the vertex ID associated to a dart.
+// --- icell-related code
+impl<T: CoordsFloat> CMap2<T> {
+    /// Fetch vertex identifier associated to a given dart.
     ///
     /// # Arguments
     ///
-    /// - `dart_id: DartIdentifier` -- ID of the dart of interest.
-    /// - `vertex_id: VertexIdentifier` -- Unique vertex identifier.
+    /// - `dart_id: DartIdentifier` -- Identifier of *dart*.
     ///
-    /// # Example
+    /// # Return / Panic
     ///
-    /// See [CMap2] example.
+    /// Return the identifier of the associated vertex.
     ///
-    pub fn set_vertexid(&mut self, dart_id: DartIdentifier, vertex_id: VertexIdentifier) {
-        self.dart_data.associated_cells[dart_id as usize].vertex_id = vertex_id;
+    pub fn vertex_id(&self, dart_id: DartIdentifier) -> VertexIdentifier {
+        Orbit2::new(self, OrbitPolicy::Vertex, dart_id)
+            .min()
+            .unwrap() as VertexIdentifier
     }
 
-    /// Set the face ID associated to a dart.
+    /// Fetch edge associated to a given dart.
     ///
     /// # Arguments
     ///
-    /// - `dart_id: DartIdentifier` -- ID of the dart of interest.
-    /// - `face_id: FaceIdentifier` -- Unique face identifier.
+    /// - `dart_id: DartIdentifier` -- Identifier of *dart*.
     ///
-    /// # Example
+    /// # Return / Panic
     ///
-    /// See [CMap2] example.
+    /// Return the identifier of the associated edge.
+    ///
+    pub fn edge_id(&self, dart_id: DartIdentifier) -> EdgeIdentifier {
+        Orbit2::new(self, OrbitPolicy::Edge, dart_id).min().unwrap() as EdgeIdentifier
+    }
```

Comment:
Because the identifier types are aliases (not structs) and the aliases point to the same primitives, `clippy` detect it as a useless type conversion & throws a warning. This might actually be an inconsistent behavior of the linter; I'll look into it.